iothub/device/samples/PnpDeviceSamples/Thermostat/Parameter.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using CommandLine;
+
+namespace Thermostat
+{
+    /// <summary>
+    /// Parameters for the application supplied via command line arguments. It default to environment variables..
+    /// </summary>
+    internal class Parameters
+    {
+        [Option(
+            's',
+            "DeviceProvisioningType",
+            HelpText = "(Required) The provisioning type that will be used for provisioning the device for the sample. Possible values include dps, hubConnectionString (case-insensitive)." +
```

Comment:
Cannot set the option as "required", since then we cannot dynamically resolve the value if it needs to be sent via environment variables. This is a trade-off in the sample supporting both env var and command line args.